library/core/src/fmt/mod.rs

```diff
@@ -2340,6 +2340,11 @@ impl Debug for str {
         f.write_char('"')?;
         let mut from = 0;
         for (i, c) in self.char_indices() {
+            // a fast path for ASCII chars that do not need escapes:
+            if matches!(c, ' '..='~') && !matches!(c, '\\' | '\"') {
```

Comment:
Do we want to use https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_graphic or some combination of other methods rather than hard-coding this?

Reply:
`is_ascii_graphic` does not *include* space, and I don’t think `\` or `"` are part of any of the other methods?

Reply:
There's `escape_ascii` but it escapes more than those two.

Reply:
The loop doesn't look like it would vectorize or even unroll well. For a string of chars that don't need escaping it might be worth processing the utf8 bytes directly [like EscapeAscii](https://github.com/rust-lang/rust/blob/7d3702e472b99be0f5de6608dd87af1df8f99428/library/core/src/slice/ascii.rs#L268-L280)